mole/models/path.py

```diff
@@ -44,36 +43,48 @@ def lessThan(
-        left_is_path = self.sourceModel().data(left, IS_PATH_ITEM_ROLE)
-        right_is_path = self.sourceModel().data(right, IS_PATH_ITEM_ROLE)
-
-        # Header items should come before path items
-        if left_is_path != right_is_path:
-            return right_is_path
-
-        # Get the values stored in UserRole for proper type comparison
-        left_data = self.sourceModel().data(left, qtc.Qt.UserRole)
-        right_data = self.sourceModel().data(right, qtc.Qt.UserRole)
-
-        # If we have UserRole data available, use it for comparison
-        if left_data is not None and right_data is not None:
-            # Compare based on the actual types
-            return left_data > right_data
-
-        # Fall back to string comparison of display text
-        left_text = self.sourceModel().data(left, qtc.Qt.DisplayRole)
-        right_text = self.sourceModel().data(right, qtc.Qt.DisplayRole)
-
-        # Try numeric comparison first if both are convertible to numbers
-        try:
-            left_num = float(left_text)
-            right_num = float(right_text)
-            return left_num > right_num
-        except Exception as _:
-            pass
-        # Fall back to string comparison
-        return str(left_text).lower() > str(right_text).lower()
+        # Access model data
+        model = self.sourceModel()
+        lft_id = model.data(left, PathRole.ID.index)
+        lft_level = model.data(left, PathRole.LEVEL.index)
+        lft_sort = model.data(left, PathRole.SORT.index)
+        lft_text = str(model.data(left, qtc.Qt.DisplayRole)).lower()
+        rgt_id = model.data(right, PathRole.ID.index)
+        rgt_level = model.data(right, PathRole.LEVEL.index)
+        rgt_sort = model.data(right, PathRole.SORT.index)
+        rgt_text = str(model.data(right, qtc.Qt.DisplayRole)).lower()
+        # Left is header
+        if lft_id is None:
+            # Right is header
+            if rgt_id is None:
+                try:
+                    lft_level = int(lft_level)
+                    rgt_level = int(rgt_level)
+                except Exception as _:
+                    if lft_text == rgt_text:
+                        return False
+                    return lft_text >= rgt_text
+                if lft_level == rgt_level:
+                    return False
+                return lft_level >= rgt_level
+            # Right is path
+            return True
+        # Left is path
+        else:
+            # Right is header
+            if rgt_id is None:
+                return False
+            # Right is path
+            try:
+                lft_sort = int(lft_sort)
+                rgt_sort = int(rgt_sort)
+            except Exception as _:
+                if lft_text == rgt_text:
+                    return False
+                return lft_text >= rgt_text
+            if lft_sort == rgt_sort:
+                return False
+            return lft_sort >= rgt_sort
```

Comment:
I'm not sure I understand this, the code as it was before seems more developer-friendly to me. Also the if/else seems to be quite duplicate maybe you can use an helper function?